services/gitlab/gitlab-stars.tester.js

```diff
@@ -0,0 +1,20 @@
+import { isMetric } from '../test-validators.js'
+import { createServiceTester } from '../tester.js'
+
+export const t = await createServiceTester()
+
+t.create('Stars')
+  .get('/gitlab-org/gitlab.json')
+  .expectBadge({
+    label: 'stars',
+    message: isMetric,
+    color: 'blue',
+    link: ['https://gitlab.com/gitlab-org/gitlab'],
+  })
+
+t.create('Stars (project not found)')
+  .get('/user1/gitlab-does-not-have-this-repo.json')
+  .expectBadge({
+    label: 'stars',
+    message: 'project not found',
+  })
```

Comment:
Similar comment to https://github.com/badges/shields/pull/8208#discussion_r927078542

It would be great if we had tests for the private repo and custom `baseUrl` cases.